### PR TITLE
OCaml 4.12

### PR DESCRIPTION
**Pull request by leahneukirchen**
First stab at it.

Currently broken:

- [x] xen
- [x] libguestfs
- [x] unison
- ~~polygen~~
- [x] coccinelle

### CODE REVIEW

**Comment by ahesford** [comment]
Unison has been patched: https://github.com/bcpierce00/unison/commit/2e12fc4f41b39ca9a5e501d7df5c2c33a75d3b95

File: srcpkgs/ocaml-findlib/template [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Template file for 'ocaml-findlib'
 pkgname=ocaml-findlib
 version=1.8.1
```
Comment (kit-ty-kate) [inline srcpkgs/ocaml-findlib/template:3]: 1.9.1 has been released (when used the build dep on `m4` can be removed)

**Comment by kit-ty-kate** [comment]
coccinelle.1.1.0 is compatible: https://github.com/ocaml/opam-repository/pull/18249

**Comment by kit-ty-kate** [comment]
`polygen` is being fixed here: https://github.com/alvisespano/Polygen/pull/22

**Comment by kit-ty-kate** [comment]
> Unison has been patched: [bcpierce00/unison@2e12fc4](https://github.com/bcpierce00/unison/commit/2e12fc4f41b39ca9a5e501d7df5c2c33a75d3b95)

Said patch is included in [2.51.4~rc2](https://github.com/bcpierce00/unison/releases/tag/v2.51.4_rc2)

**Comment by kit-ty-kate** [comment]
`libguestfs` 1.44.1 should support OCaml 4.12: https://download.libguestfs.org/1.44-stable/libguestfs-1.44.1.tar.gz

**Comment by leahneukirchen** [comment]
@thypon I'm tempted to remove polygen, it's upstream unmaintained and really not very useful...

**Comment by thypon** [comment]
Go for it. It is just a toy project anyway. I'm not sure anybody uses it in voidlinux besides me. Package popularity-context or download count could had helped.

**Comment by leahneukirchen** [comment]
Polygen has popcorn = 1.

**Pull request merged**